BibBuddy/app/src/main/java/de/bibbuddy/BookBarcodeScannerFragment.java

```diff
@@ -131,28 +132,40 @@ public void receiveDetections(Detector.Detections<Barcode> detections) {
   }
 
   private void handleIsbnInput(String isbn) {
-    isbnRetriever = new IsbnRetriever(isbn);
-    System.out.println(isbnRetriever);
-    thread = new Thread(isbnRetriever);
-    thread.start();
-
-    try {
-      thread.join();
-    } catch (Exception e) {
-      System.out.println(e);
-    }
+    String cleanIsbn = isbn.replaceAll("\\s", "");
+
+    if (dataValidation.isValidIsbn10or13(cleanIsbn)) {
+      isbnRetriever = new IsbnRetriever(cleanIsbn);
+      System.out.println(isbnRetriever);
```

Comment:
```suggestion
```
If you do not need that system-print, you should remove it - otherwise: nevermind.

Reply:
Done.